pkg/build/cleanup.go

```diff
@@ -11,8 +11,16 @@ import (
 // temporary directories created by STI build and it also cleans the temporary
 // Docker images produced by LayeredBuild
 type DefaultCleaner struct {
-	util.FileSystem
-	docker.Docker
+	fs     util.FileSystem
+	docker docker.Docker
```

Comment:
I would actually argue (but reasonable minds are welcome to differ) that the primary reason to embed one struct in another is to effectively make the parent struct be able to serve as (be passed as) the child struct.  It's composition's equivalent to inheritance.  So in the case where you embed the Mutex above, i don't think that's actually a good reason to do it, unless you have a need to pass Counter to functions that take Mutexes, just like in an inherited language you probably wouldn't make Counter extend or implement Mutex (though you might, certainly).  If you just need a Mutex on Counter, then making the mutex a field and calling "c.mutex.Lock()" seems reasonable to me, though I won't argue w/ the convenience embedding gives you to just call c.Lock().

At the end of the day though, I would agree that the default preference should be to *not* embed things.